driver: wifi: esp_offload.c: Missing new timeout API conversion

**Describe the bug**
The esp_offload.c don't build because are missing some new timeout API conversions.

**To Reproduce**
west build -b frdm_k64f samples/net/wifi -- -DSHIELD=esp_8266

**Expected behavior**
Build as usual

**Screenshots or console output**
```
/zephyr/drivers/wifi/esp/esp_offload.c
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c: In function 'esp_connect':
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:148:14: error: invalid operands to binary == (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
  148 |  if (timeout == K_NO_WAIT) {
      |              ^~
In file included from ../include/kernel_includes.h:41,
                 from ../include/kernel.h:17,
                 from ../include/zephyr.h:18,
                 from /home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:11:
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c: In function 'esp_sendto':
../include/sys_clock.h:85:28: error: incompatible type for argument 5 of 'esp_connect'
   85 | #define Z_TIMEOUT_TICKS(t) ((k_timeout_t) { .ticks = (t) })
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            |
      |                            k_timeout_t {aka struct <anonymous>}
../include/sys_clock.h:87:25: note: in expansion of macro 'Z_TIMEOUT_TICKS'
   87 | #define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS(k_ms_to_ticks_ceil32(MAX(t, 0)))
      |                         ^~~~~~~~~~~~~~~
../include/kernel.h:1585:24: note: in expansion of macro 'Z_TIMEOUT_MS'
 1585 | #define K_MSEC(ms)     Z_TIMEOUT_MS(ms)
      |                        ^~~~~~~~~~~~
../include/kernel.h:1597:24: note: in expansion of macro 'K_MSEC'
 1597 | #define K_SECONDS(s)   K_MSEC((s) * MSEC_PER_SEC)
      |                        ^~~~~~
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:361:8: note: in expansion of macro 'K_SECONDS'
  361 |        K_SECONDS(5), NULL);
      |        ^~~~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:124:16: note: expected 's32_t' {aka 'int'} but argument is of type 'k_timeout_t' {aka 'struct <anonymous>'}
  124 |          s32_t timeout,
      |          ~~~~~~^~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:377:14: error: invalid operands to binary == (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
  377 |  if (timeout == K_NO_WAIT) {
      |              ^~
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c: In function 'esp_recv':
/home/gfbudke/zephyros/mainline/zephyr/drivers/wifi/esp/esp_offload.c:585:14: error: invalid operands to binary == (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
  585 |  if (timeout == K_NO_WAIT) {
      |              ^~
[66/176] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/timer/sys_clock_init.c.obj
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.11.2
 - Commit SHA ed2d263

--- a/drivers/wifi/esp/esp_offload.c
+++ b/drivers/wifi/esp/esp_offload.c
@@ -145,7 +145,7 @@ static int esp_connect(struct net_context *context,
 	sock->connect_cb = cb;
 	sock->conn_user_data = user_data;
 
-	if (timeout == K_NO_WAIT) {
+	if (timeout == 0) {
 		k_work_submit_to_queue(&dev->workq, &sock->connect_work);
 		return 0;
 	}
@@ -358,7 +358,7 @@ static int esp_sendto(struct net_pkt *pkt,
 			 * have a valid link id before proceeding.
 			 */
 			ret = esp_connect(context, dst_addr, addrlen, NULL,
-					  K_SECONDS(5), NULL);
+					  (5 * MSEC_PER_SEC), NULL);
 			if (ret < 0) {
 				return ret;
 			}
@@ -374,7 +374,7 @@ static int esp_sendto(struct net_pkt *pkt,
 	sock->send_cb = cb;
 	sock->send_user_data = user_data;
 
-	if (timeout == K_NO_WAIT) {
+	if (timeout == 0) {
 		k_work_submit_to_queue(&dev->workq, &sock->send_work);
 		return 0;
 	}
@@ -582,7 +582,7 @@ static int esp_recv(struct net_context *context,
 	sock->recv_user_data = user_data;
 	k_sem_reset(&sock->sem_data_ready);
 
-	if (timeout == K_NO_WAIT) {
+	if (timeout == 0) {
 		return 0;
 	}
 

